source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -88,29 +103,88 @@ GeoipProvider::~GeoipProvider() {
   ENVOY_LOG(debug, "Shutting down Maxmind geolocation provider");
   if (city_db_) {
     MMDB_close(city_db_.get());
+    city_db_ = nullptr;
   }
   if (isp_db_) {
     MMDB_close(isp_db_.get());
+    isp_db_ = nullptr;
   }
   if (anon_db_) {
     MMDB_close(anon_db_.get());
+    anon_db_ = nullptr;
+  }
+  if (country_mapping_) {
```

Comment:
why this is `if` block is needed?

Reply:
country_mapping_ is a pointer as anon_db_ and others. keeping the same standard

Reply:
country_mapping_ is smart pointer to C++ `absl::map`, there is no need to clean it within the destructor of the owning object (provider). While all db types are smart pointers to C struct types, in C there are no destructors and maxmind library recommends to call `MMDB_close` to release dynamically allocated memory.